oxide_core/test/vector/vector3.rs

```diff
@@ -0,0 +1,137 @@
+        let tar: f32 = 50.0;
+        let res = new_vector.get_squared_length();
+        assert_approx_eq!(tar, res);
+    }
+
+    #[test]
+    fn get_length() {
+        let new_vector = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let tar: f32 = 7.071_068;
+        let res = new_vector.get_length();
+        assert_approx_eq!(tar, res);
+    }
+
+    #[test]
+    fn inner_product() {
+        let new_vector_1 = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let new_vector_2 = vector3::Vector3 {
+            x: 3.0,
+            y: 7.0,
+            z: 11.0,
+        };
+        let tar: f32 = 92.0;
+        let res = new_vector_1.inner_product(&new_vector_2);
+        assert_approx_eq!(tar, res);
+    }
+
+    #[test]
+    fn scale_3() {
+        let new_vector = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let tar_vector = vector3::Vector3 {
+            x: 9.0,
+            y: 12.0,
+            z: 15.0,
+        };
+        let res_vector = new_vector.scale(3.0);
+        assert_vector3_approx_eq(&tar_vector, &res_vector);
+    }
+
+    #[test]
+    fn negate() {
+        let new_vector = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let tar_vector = vector3::Vector3 {
+            x: -3.0,
+            y: -4.0,
+            z: -5.0,
+        };
+        let res_vector = new_vector.negate();
+        assert_vector3_approx_eq(&tar_vector, &res_vector);
+        assert_approx_eq!(tar_vector.get_length(), res_vector.get_length());
+    }
+
+    #[test]
+    fn add() {
+        let new_vector_1 = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let new_vector_2 = vector3::Vector3 {
+            x: 3.0,
+            y: 7.0,
+            z: 11.0,
+        };
+        let tar_vector = vector3::Vector3 {
+            x: 6.0,
+            y: 11.0,
+            z: 16.0,
+        };
+        let res_vector = new_vector_1 + new_vector_2;
+        assert_vector3_approx_eq(&tar_vector, &res_vector);
+    }
+    #[test]
+    fn sub() {
+        let new_vector_1 = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let new_vector_2 = vector3::Vector3 {
+            x: 3.0,
+            y: 7.0,
+            z: 11.0,
+        };
+        let tar_vector = vector3::Vector3 {
+            x: 0.0,
+            y: -3.0,
+            z: -6.0,
+        };
+        let res_vector = new_vector_1 - new_vector_2;
+        assert_vector3_approx_eq(&tar_vector, &res_vector);
+    }
+
+    #[test]
+    fn normalize() {
+        let new_vector = vector3::Vector3 {
+            x: 3.0,
+            y: 4.0,
+            z: 5.0,
+        };
+        let tar_vector = vector3::Vector3 {
+            x: 0.42426407871192851464050661726291,

```

Comment:
Due to precision limit, ```0.42426407871192851464050661726291``` evaluates to ```0.4242640787119285```.
```suggestion
            x: 0.424_264_07,
```

Reply:
I know it's more than its precision, but I've been thinking about when the implementation changes in the future (for example, when the implementation changes to f64), what would be better?

Reply:
If there is an occasion to change the type to "f64", I think the test code should be changed accordingly. The current type is "f32", so the test code should be tailored as well.

Reply:
@hyp3rflow What do you think?

Reply:
> The current type is "f32", so the test code should be tailored as well.  

I agree with you 😄

Reply:
I also agree with this change.